### DISCUSSION TITLE
arrayField 中的字段绑定主动联动，在特定场景下联动失效无法触发

### DISCUSSION THREAD

**Discussion by duhongjun** [alibaba/formily#3630] [Q&A 2.x]
### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/stupefied-voice-4eoouy?file=/App.tsx)

### Steps to reproduce
1. 点击新增，创建出来三个array item，其中的input值发生变化会触发一个reaction， 效果是alert一下
2. 分别修改这三个input的内容，触发这三个array item中 input的联动，此时都是正常的；
3. 删除第二个array item， 此时之前的第三个array item会变成新的第二个array item，此时修改它的input的值，联动失效，无法触发。

### What is expected?
正常触发联动。

### What is actually happening?
无法触发联动

### Package
@formily/react@2.2.14

---

**Comment by duhongjun** [comment]
我的场景是A和其他字段有联动关系，A字段发生变化时将最新的值通过setComponentProps传入到指定的字段中

**Reply by dddddd889** [reply]
你可以尝试下使用
```
const form = createForm({
  effects() {
    onFieldValueChange('string_array.*', (field) => {
			alert(`${field.path.entire} changed!`)
    })
  },
})
```
